### DISCUSSION TITLE
v3 Feedback/Questions

### DISCUSSION THREAD

**Discussion by thejoyfuldev** [chakra-ui/chakra-ui#9006] [General]
Few thoughts.

Do I have to use snippets? The reason I use ChakraUI is so I don't have to manage components myself. If the CLI will not update previously added files, that means the upkeep of components has now been offloaded.

When I add a new component, I need to spend more time ensuring the generated component file aligns with my tooling.
- Several components have a `use client` at the top which is only required in NextJS.
- Vite uses `ts-eslint` which enables the `no-empty-object-type` by default. Several components break this rule which immediately introduces linting errors into my codebase which I now have to fix.
```tsx
interface ColorModeButtonProps extends Omit<IconButtonProps, "aria-label"> {}
```

Since the props is empty, it should be defined as:
```tsx
type ColorModeButtonProps = Omit<IconButtonProps, "aria-label">
```
- Vite uses React Refresh ESLint plugin which throws a warning because several components are exporting functions (i.e. hooks) in the same file as components.
- Since the CLI doesn't update previously generate components, I now have to manually update each one even if nothing in the file has changed.
- While I like the idea of being able to use different icons, it seems to be a tedious task to do so. I think giving the icons specific names would speed this process up. So instead of keeping the original icon name, `LuCheck` you could import/rename the icon to `CheckIcon` so that when we swap the icon library, it's just as simple as `import { CheckIcon } from "lucide-react"` or something similar.

I don't know the first thing about Chakra codebase and how it aligns with Ark UI (saw it mentioned elsewhere), but Chakra v3 seems it could be split into two different packages: 1) Components, 2) Blocks (aka snippets). If I just want to use the standard Chakra Button, I just use the default component. However, if I want to have a Checkbox Card, then I could use the snippet cli to add that pre-built component.

Will v2 still be maintained? For most of my projects, I just need OOB components that work. I really don't need the overhead of adding a snippet, fixing linting issues, then updating regularly as changes occur, etc. Of am I missing something here?

**Comment by bbovenzi** [comment]
+1 on making snippets a separate package. Adding them and making them work with vite and our linting/typing rules was very tedious. This tedium will continue as chakra changes, I need to maintain all these components. It would be much easier to import all snippets from a maintained library and only create custom components when I need to.

**Comment by jrtnq514** [comment]
Also please update the v3 docs to provide vanilla examples of components instead of just the snippets.

**Comment by vitalijalbu** [comment]
Another thing i would like to discuss are button truggers, sometimes we have external trigger and need to use **isOpen props**, but it's missing in the docs...

**Comment by veloware** [comment]
I have the same concerns as some of the above comments, regarding the upkeep of snippets once I add them into my code base, is it not possible to just use `import { Button } from '@chakra-ui/react'` anymore?

If the components now get _copied_ into my project rather than just referenced from a package, what is the benefit of using Chakra over just creating my own components using Ark?

(long time user of Chakra here, and have loved using the project so far, but if my above assumption are correct, I feel a little uneasy using it now because of the potential upkeep of external components)

**Comment by veloware** [comment]
@j-bullard I see you are able to turn that rule off when its being used for *Prop types, something like this has worked for me so far - 

```
"rules": {
    "@typescript-eslint/no-empty-object-type": [
      2,
      {
         "allowWithName": "Props$"
      }
    ]
  }
```
